### ISSUE TITLE
Will universal-router be maintained in the near future?

### ISSUE THREAD

**Issue by adrian-moisa** [kriasoft/universal-router#135]
Sounds like a dummy question but let's review some context. I have been working with angularjs, angular2 and react in the previous years and all of them have popular routing solutions. Now I'm trying to ditch frameworks in favor of native web components. So far I managed to build a basic demo. Now I need to setup a proper routing solution. I've been looking around for possible solutions and I have found [Visionmedia page](https://github.com/visionmedia/page.js) and [Flatiron director])(https://github.com/flatiron/director). Both of them seem to have been abandoned by the maintainers. for more than 2 years. I am looking for a router that will be around for the foreseeable future. 

So basically I need some reassurance about the future of universal-router. The project I build I expect it to be alive for several years. Looking at the contribution graphs it looks like the repository is really healthy.

**Comment by frenzzy** [comment]
Hey @adrian-moisa, thank you for your interest. This project is maintained since Mid 2015 by @kriasoft org and used in popular projects such as [React Starter Kit](https://github.com/kriasoft/react-starter-kit) (15k stars) and [React Static Boilerplate](https://github.com/kriasoft/react-static-boilerplate) (3k stars) supported by the same maintainer. These projects are alive and actively used in real-life products. So yes, I think it will be maintained in the near future.

Upcoming project updates: add more [recipes](https://github.com/kriasoft/universal-router/pull/86), improve [website](https://www.kriasoft.com/universal-router/) and maybe implement an option for custom global error handler.

**Comment by adrian-moisa** [comment]
Thank you for the quick answer. I have already started implementing the library. So it looks like I'm going to stick around. 👍

**Comment by chrismbeckett** [comment]
+1 - also moving to native web components and looking for a good router solution. Nice to see this solution will be maintained into the future.

**Comment by adrian-moisa** [comment]
@chrismbeckett I will publish soon a npm module that bridges the gap between universal router and web components. Currently if you use universal router together with web components as indicated in the  [Quickstart tutorial, usage with React](http://slides.com/koistya/universal-router#/7) you will have lots of trouble with rendering the routes in an efficient way. Due to the fact that react uses a virtual dom the update of the view is already efficient by default. 

For webcomponents it is necessary to use a `router-outlet` component that updates only the newsted web components. For this you need to generate a tree of rotues using universal router and than parse that tree to acutally update teh web components tree in an efficient way. Right now I'm working on a prototype that fixes this issue. I will publish in on github as soon as I have a stable version.

I'm also working on nano data binding solution for web components that will remove all the boilerplate involved in setting up nested web components in large scale apps. It will have the typical DSL that you see in an angualr app but using far less code (400 LOC so far). My intent is to provide a data binding solution for web components that anybody can understand ende to end in 30 mins of study.

I'll keep you updated as soon as I have published a minimal working version.

**Comment by Christian24** [comment]
@adrian-moisa that sounds awesome. Any update on that?

**Comment by fehmi** [comment]
Looks like it is not possible...
